### PR TITLE
Feature/svg spritemap builder

### PR DESCRIPTION
**Pull request by Jaervinen**
Add support to build SVG spritemaps from single SVG files inside a folder / folder-structure

### CODE REVIEW

File: lib/templates/conventional.config.json [hunk 1]
```diff
@@ -16,5 +16,11 @@
         "outputPath": "./Resources/Public/JavaScript/",
         "outputFilePattern": "[name].min.js"
     },
-    "browserUpdateNote": false
+    "browserUpdateNote": false,
+    "spritemap": {
+        "prefix": "test-",
+        "inputPath": "./Resources/Public/Images/SVG-icons/**/*.svg",
```
Comment (s2b) [inline lib/templates/conventional.config.json]: To stay consistent, I would suggest to use `inputFiles` instead, where each file can contain glob syntax

File: lib/templates/conventional.config.json [hunk 2]
```diff
@@ -16,5 +16,11 @@
         "outputPath": "./Resources/Public/JavaScript/",
         "outputFilePattern": "[name].min.js"
     },
-    "browserUpdateNote": false
+    "browserUpdateNote": false,
+    "spritemap": {
+        "prefix": "test-",
```
Comment (s2b) [inline lib/templates/conventional.config.json]: Should this be the default value?

File: lib/webpackBuilder.js [hunk 3]
```diff
@@ -4,6 +4,10 @@ const availableTypes = [
 ]
 
 const build = (type, mode, workingDir, config) => {
+  if (config.spritemap) {
```
Comment (s2b) [inline lib/webpackBuilder.js]: we should add a generic approach to disable certain features. Maybe something like this:

```json
"sass": {
  "enabled": true
}
```

File: lib/templates/conventional.config.json (hunk 2 above)
Comment (Jaervinen) [reply lib/templates/conventional.config.json]: No, the "prefix" config is optional and can be omitted, default is therefore empty. I believed that this json template should show all possible config options?

Comment (s2b) [reply lib/templates/conventional.config.json]: Yes, but it is also the default template for new projects (copied via `conventional init`). I would suggest to leave the `prefix` in but to assign an empty string.

By the way: This feature should be mentioned in README.md

File: lib/templates/conventional.config.json (hunk 1 above)
Comment (Jaervinen) [reply lib/templates/conventional.config.json]: The plugin we use can, so far, only accept 1 glob path. I changed the option name anyway.

**Comment by s2b** [comment]
Build returns the following notice:

```
"extendDefaultPlugins" utility is deprecated.
Use "preset-default" plugin with overrides instead.
For example:
{
  name: 'preset-default',
  params: {
    overrides: {
      // customize plugin options
      convertShapeToPath: {
        convertArcs: true
      },
      // disable plugins
      convertPathData: false
    }
  }
}
```

Can this be fixed easily?

**Comment by s2b** [comment]
`enabled` flag wasn't such a good idea after all. :) Could you try to interpret the existing options to determine automatically if that type is enabled?

**Comment by Jaervinen** [comment]
> Build returns the following notice:
> 
> ```
> "extendDefaultPlugins" utility is deprecated.
> Use "preset-default" plugin with overrides instead.
> For example:
> {
>   name: 'preset-default',
>   params: {
>     overrides: {
>       // customize plugin options
>       convertShapeToPath: {
>         convertArcs: true
>       },
>       // disable plugins
>       convertPathData: false
>     }
>   }
> }
> ```
> 
> Can this be fixed easily?

This is fixed in the recent commit with upgrading the svg-spritemap-webpack-plugin

> Build returns the following notice:
> 
> ```
> "extendDefaultPlugins" utility is deprecated.
> Use "preset-default" plugin with overrides instead.
> For example:
> {
>   name: 'preset-default',
>   params: {
>     overrides: {
>       // customize plugin options
>       convertShapeToPath: {
>         convertArcs: true
>       },
>       // disable plugins
>       convertPathData: false
>     }
>   }
> }
> ```
> 
> Can this be fixed easily?

Yes – this is fixed in the recent commit by upgrading svg-spritemap-webpack-plugin

**Comment by Jaervinen** [comment]
> Build returns the following notice:
> 
> ```
> "extendDefaultPlugins" utility is deprecated.
> Use "preset-default" plugin with overrides instead.
> For example:
> {
>   name: 'preset-default',
>   params: {
>     overrides: {
>       // customize plugin options
>       convertShapeToPath: {
>         convertArcs: true
>       },
>       // disable plugins
>       convertPathData: false
>     }
>   }
> }
> ```
> 
> Can this be fixed easily?

Yes – this is fixed in the recent commit by upgrading svg-spritemap-webpack-plugin

**Pull request merged**